src/wallet.ts

```diff
@@ -0,0 +1,2170 @@
+
+    amountInput.key('enter', function() {
+        transferButton.press();
+    });
+
+    amountInput.key(['escape'], function() {
+        screen.focusPop();
+        transferWindow.focus();
+    })
+
+    amountInput.key(['tab'], function() {
+        screen.focusPop();
+        transferWindow.focus();
+    })
+
+    amountInput.on('blur', function() {
+        transferWindow.focus();
+    })
+
+    // pop focus on textbox click
+    idInput.on('click', function() {
+        screen.focusPop();
+    });
+
+    let availableBalanceText = blessed.text({
+        parent: transferForm,
+        top: 9,
+        left: 21,
+        fg: 'white',
+        tags: true,
+        content: `{bold}Available:{/}\n${WB.prettyPrintAmount(walletBalanceData[0])}`
+    })
+
+    // define submit button
+    let transferButton = blessed.button({
+        parent: transferForm,
+        mouse: true,
+        shrink: true,
+        padding: {
+            left: 14,
+            right: 14
+        },
+        left: 0,
+        top: 12,
+        content: 'send (enter)',
+        style: {
+            bg: 'black',
+            fg: 'white',
+            hover: {
+                bg: 'red'
+            }
+        }
+    });
+
+    // on send transfer button press
+    transferButton.on('press', function() {
+        screen.focusPop();
+        transferForm.submit();
+        transferForm.reset();
+    });
+
+    // on transaction
+    wallet.on('transaction', async function() {
+
+        notifyUser(notificationText, 'New transation found!', 2000);
+
+        // get wallet data and set it in walletBalance
+        let updateBalance = wallet.getBalance();
+        walletBalance.setContent(
+            `{|}{bold}${WB.prettyPrintAmount(updateBalance[0])}{/}\n` +
+            `{|}{bold}{red-fg}${WB.prettyPrintAmount(updateBalance[1])}{/}\n` +
+            `{|}${WB.prettyPrintAmount(updateBalance[1] + updateBalance[0])}`);
+
+        let updateTransactionList = wallet.getTransactions();
+
+        const updateTxArray = getRecentTransactions(updateTransactionList);
+
+        transactionTable.setData({
+            headers: ['Time', 'Amount'],
+            data: updateTxArray
+        })
+
+        // render the screen
+        screen.render();
+    })
+
+    // focus window and render screen
+    walletWindow.focus();
+    screen.render();
+
+    // return undefined if no error
+    return undefined;
+};
+
+// create new wallet and launch it
+function createWallet(fileName, password) {
+    if (fs.existsSync(walletDirectory + `/${fileName}.wallet`)) {
+        return `${fileName}.wallet already exists, will not overwrite.`;
+    };
+    const wallet = WB.WalletBackend.createWallet(daemon);
+    wallet.saveWalletToFile(walletDirectory + `/${fileName}.wallet`, password);
+    wallet.stop();
+    drawWalletWindow(fileName, password);
+};
+
+// imports a wallet by seed and launch it
+function importWallet(fileName: string, password: string, seed: string, startHeight: number) {
```

Comment:
Nice typings, now add a return type
```suggestion
function importWallet(fileName: string, password: string, seed: string, startHeight: number): WalletError | undefined {
```